common/src/main/java/com/lightstep/tracer/shared/Span.java

```diff
@@ -84,11 +76,8 @@ public Span setTag(String key, Number value) {
             return this;
         }
         synchronized (mutex) {
-            if (isJoinKey(key)) {
-                record.addToJoin_ids(new TraceJoinId(key, value.toString()));
-            } else {
-                record.addToAttributes(new KeyValue(key, value.toString()));
-            }
+            // TODO convert to number value? lose precision?
```

Comment:
Yeah, we'll want to maintain the Number-ness of numeric tags as long as `KeyValue`'s value looks like a variadic type. Same with bools.

Ironically, everything is coerced to a string in the collector, but that's not the endgame.